lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2847,7 +2847,7 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
                 Nothing -> []
 
         unbalancedTx <- liftHandler $
-            W.constructTransaction @n era
+            W.constructTransaction @n @era
```

Comment:
@Anviking Thanks for looking at this.

> If `@era` is required to avoid ambiguity I don't think it's the right way to go 🤔

Can you suggest an alternative?

We already use this `@era` type parameter style in many places:
```hs
lib/wallet/api/http/Cardano/Wallet/Api/Http/Server/Error.hs:623:                , candidateTxHex = hexText $ Write.serializeTx @era candidateTx
lib/wallet/src/Cardano/Wallet.hs:2207:        buildTransactionPure @s @era
lib/wallet/src/Cardano/Wallet.hs:2279:    anyCardanoEra = Cardano.AnyCardanoEra $ Write.cardanoEra @era
lib/wallet/src/Cardano/Wallet.hs:2309:            buildTransactionPure @s @era
lib/wallet/src/Cardano/Wallet.hs:2345:            mkUnsignedTransaction (recentEra @era)
lib/wallet/src/Cardano/Wallet.hs:2351:            Write.constructUTxOIndex @era $
lib/wallet/src/Cardano/Wallet.hs:2989:                $ Write.constructUTxOIndex @era
lib/wallet/src/Cardano/Wallet.hs:3007:                first (Write.toCardanoApiTx @era) <$>
lib/wallet/src/Cardano/Wallet.hs:3022:                            -> case Write.recentEra @era of {}
lib/wallet/src/Cardano/Wallet.hs:3027:    era = recentEra @era
lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:313:        mkUnsignedTx @era
lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:368:                @era
lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:440:    era = Write.recentEra @era
lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:1222:    era = Write.recentEra @era
lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:998:            result <- runExceptT (calculateFeePercentiles @_ @era estimateFee)
...
```

At the moment, it seems as though we are currently using a mixture of the following styles for function type signatures:

 - style 1: only `RecentEra` value
 - style 2: only `IsRecentEra` constraint
 - style 3: both `IsRecentEra` constraint and `RecentEra` value
 
I think it would be beneficial if we could come to some kind of consensus about which style to use. What do you think?

Reply:
I'll think about writing something in the docs for `IsRecentEra` / `RecentEra`, apologies for not getting to it earlier